src/storage/useFilePreview.ts

```diff
@@ -0,0 +1,60 @@
+'use client'
+
+import { useQuery, useQueryClient, UseQueryOptions } from '@tanstack/react-query'
+import { useMemo } from 'react'
+import { useAppwrite } from 'react-appwrite'
+
+type ImageOptionsProp = {
+  width?: number,
+  height?: number,
+  gravity?: number,
+  quality?: number,
+  borderWidth?: number,
+  borderColor?: string,
+  borderRadius?: number,
+  opacity?: number,
+  rotation?: number,
+  background?: string,
+  output?: string,
+}
+
+/**
+ * Access to the file object by its unique file ID.
+ * @param bucketId Storage bucket unique ID
+ * @param fileId  File ID
+ * @param imageOptions Options for cutting and resizing your preview image.
+ * @param options Options to pass to `react-query`
+ * @link [Appwrite Documentation](https://appwrite.io/docs/client/storage?sdk=web-default#storageGetFilePreview)
+ */
+export function useFilePreview(
+  bucketId: string,
+  fileId: string,
+  imageOptions: ImageOptionsProp = {
+    width: undefined,
+    height: undefined,
+    gravity: undefined,
+    quality: undefined,
+    borderWidth: undefined,
+    borderColor: undefined,
+    borderRadius: undefined,
+    opacity: undefined,
+    rotation: undefined,
+    background: undefined,
+    output: undefined,
+  },
+  options?: UseQueryOptions<URL, unknown, URL, string[]>
+) {
+  const { storage } = useAppwrite()
+  const queryClient = useQueryClient()
+  const queryKey = useMemo(() => ['appwrite', 'storage', bucketId, fileId], [bucketId, fileId])
+  const queryResult = useQuery({
+    queryKey,
+    queryFn: async ({ queryKey: [, , bucketId, fileId] }) => {
+      return await storage.getFilePreview(bucketId, fileId,  ...Object.values<any>(imageOptions))
```

Comment:
If the object is constructed with the properties out of order, the values will be passed in out of order too. We must manually pass the parameters in ourselves.

![](https://i.imgur.com/m4st0FK.png)